mcp/streamable.go

```diff
@@ -40,6 +40,8 @@ type StreamableHTTPHandler struct {
 	getServer func(*http.Request) *Server
 	opts      StreamableHTTPOptions
 
+	onTransportDeletion func(sessionID string) // for testing only
```

Comment:
I would prefer not to add this field only for testing purpose. But the test need some signal from the production indicating something have changed. 

Let me know if you have any better solution to this.